hilti/toolchain/src/compiler/validator.cc

```diff
@@ -555,6 +555,11 @@ struct VisitorPost : visitor::PreOrder, public validator::VisitorMixIn {
         }
     }
 
+    void operator()(statement::Declaration* n) final {
+        if ( ! n->declaration()->isA<declaration::LocalVariable>() )
+            error("only variables can be declared inside local scopes", n);
```

Comment:
I have a couple nitpicks here

1) Both `const` and `type` are allowed by the parser, then later rejected. I like that because they get nicer error messages (though maybe it could apply to more decls). But - the `const` check is in [Spicy's validator](https://github.com/zeek/spicy/blob/207e1cb51e5cdb970f19b802f56a43d2f26e6aba/spicy/toolchain/src/compiler/validator.cc#L320-L324) and has a different message. Maybe those can be combined into this one?
2) Since globals are variables too, it would be nice if this said locals:

```suggestion
            error("only local variables can be declared inside local scopes", n);
```

Reply:
> But - the const check is in [Spicy's validator](https://github.com/zeek/spicy/blob/207e1cb51e5cdb970f19b802f56a43d2f26e6aba/spicy/toolchain/src/compiler/validator.cc#L320-L324) and has a different message. Maybe those can be combined into this one?

Indeed, that const check can just go, it's covered by the new check. Done.

> error("only local variables can be declared inside local scopes", n);

That sounds slightly misleading to me. At least the way I read it, it would put the emphasis on "**local** variables", as a contrasts to "global variables". So leaving this, but slight tweaking it to add a "(not XXX)", with XXX being what the validator found.

Also adding a test that I forgot to commit.